apps/search/src/lib/algolia/algolia-error-parser.ts

```diff
@@ -17,4 +15,13 @@ export const AlgoliaErrorParser = {
 
     return parsed.data.status === 403 || parsed.data.status === 401;
   },
+  isRecordSizeTooBigError: (error: unknown) => {
+    const parsed = shape.safeParse(error);
+
+    if (!parsed.success) {
+      return false;
+    }
+
+    return parsed.data.status === 400 && /Record.*is too big/.test(parsed.data.message);
```

Comment:
maybe - but I'm not sure - it's better to calculate it before sending to Algolia? Like here

https://github.com/algolia/algoliasearch-magento-2/blob/3.0.1/Helper/AlgoliaHelper.php#L510-L554

I see these reasons:
- Don't rely on flaky message (these can change and likely Algolia will not consider this a breaking change) 
- No need to send it to Algolia, if we can compute record size earlier

Reply:
The record size limit is different for different plans - https://support.algolia.com/hc/en-us/articles/4406981897617-Is-there-a-size-limit-for-my-index-records